README.md

```diff
@@ -14,8 +14,44 @@ implemented in CUDA.
 Quick Install
 ------------
 
+### Prerequisites
+
+#### Linux and Mac OS 
+
+For Ginkgo core library:
+
+*   _cmake 3.9+_
+*   C++11 compliant compiler, one of:
+    *   _gcc 5.3+, 6.3+, 7.3+, 8.1+_
+    *   _clang 3.9+_
+    *   _Apple LLVM 8.0+_ (__TODO__: verify)
+
+The Ginkgo CUDA module has the following __additional__ requirements:
+
+*   _CUDA 9.0+_
+*   Any host compiler restrictions your version of CUDA may impose also apply
+    here. For the newest CUDA version, this information can be found in the
+    [CUDA installation guide for Linux](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html)
+    or [CUDA installation guide for Mac Os X](https://docs.nvidia.com/cuda/cuda-installation-guide-mac-os-x/index.html)
+
+In addition, if you want to contribute code to Ginkgo, you will also need the
+following:
+
+*   _clang-format 5.0.1+_ (ships as part of _clang_)
+
+#### Windows
+
+Windows is currently not supported, but we are working on porting the library
+there. If you are interested in helping us with this effort, feel free to
+contact one of the developers. (The library itself doesn't use any non-standard
+C++ features, so most of the effort here is in modifying the build system.)
+
+__TODO:__ Some restrictions will also apply on the version of C and C++ standard
+libraries installed on the system. We need to investigate this further.
 Use the standard cmake build procedure:
 
+To build Ginkgo, you can use the standard CMake procedure. 
```

Comment:
You should add a new section above this, maybe `### Building Ginkgo` or similar.